packages/veritone-react-common/src/components/DataTable/table.test.js

```diff
@@ -0,0 +1,696 @@
+    );
+
+    const menuItems = wrapper.find('Menu').prop('children');
+    expect(filter(menuItems, (menuItem) => includes(get(menuItem, 'key'), 'divider')).length).toEqual(0);
+  });
+});
+
+describe('Table', function () {
+  it('Renders rows provided by the rowGetter function when given column children', function () {
+    function rowGetter(i) {
+      return {
+        index: String(i),
+        name: `col-${i}`
+      };
+    };
+
+    const wrapper = mount(
+      <Table rowGetter={rowGetter} rowCount={2}>
+        <Column dataKey="index" />
+        <Column dataKey="name" />
+      </Table>
+    );
+
+    const expectedCellContents = ['0', '1', 'col-0', 'col-1'];
+
+    assertRowsExist(expectedCellContents, wrapper);
+  });
+
+  it('Renders column headers', function () {
+    const wrapper = mount(
+      <Table
+        rowGetter={() => ({})} // eslint-disable-line react/jsx-no-bind
+        rowCount={1}
+      >
+        <Column dataKey="_" header="One" />
+        <Column dataKey="_" header="Two" />
+      </Table>
+    );
+
+    const expectedHeaderContents = ['One', 'Two'];
+
+    expectedHeaderContents.forEach(expected => {
+      expect(wrapper.find('th').someWhere(n => n.text() === expected)).toEqual(
+        true
+      );
+    });
+  });
+
+  it('renders the correct row count', function () {
+    const wrapper = mount(
+      <Table
+        rowGetter={i => ({ i })} // eslint-disable-line react/jsx-no-bind
+        rowCount={10}
+      >
+        <Column dataKey="i" />
+      </Table>
+    );
+
+    expect(wrapper.find('table').find('tbody').find('tr').length).toEqual(10);
+  });
+
+  it('renders a MUI TableFooter if provided', function () {
+    const footer = (
+      <TableCell id="my-footer">
+        <div>Footer</div>
+      </TableCell>
+    );
+
+    const wrapper = mount(
+      <Table
+        rowGetter={() => ({})} // eslint-disable-line react/jsx-no-bind
+        rowCount={1}
+        footerElement={footer}
+      >
+        <Column dataKey="_" />
+      </Table>
+    );
+
+    expect(wrapper.contains(footer)).toEqual(true);
+  });
+
+  it('renders new data if data changes', function () {
+    class Container extends React.Component {
+      getRowData = i => {
+        return this.props.data[i]; // eslint-disable-line
+      };
+
+      render() {
+        return (
+          <Table rowGetter={this.getRowData} rowCount={initialData.length}>
+            <Column dataKey="id" header="id" />
+          </Table>
+        );
+      }
+    }
+
+    const initialData = [{ id: 0 }, { id: 1 }];
+
+    const wrapper = mount(<Container data={initialData} />);
+
+    wrapper.setProps(
+      {
+        data: [{ id: 0 }, { id: '1-new' }]
+      },
+      function () {
+        expect(
+          wrapper.find('td').someWhere(n => n.text() === '1-new')
+        ).toEqual(true);
+      }
+    );
+  });
+
+  it('exposes row index and column key when table cell is clicked', function () {
+    const spy = sinon.spy();
```

Comment:
should use jest mocks for these to be consistent
```javascript
const handler = jest.fn();
...
onCellClick={ handler }
...
expect(handler).toHaveBeenCalled();
```

Reply:
fixed